api/src/services/authorization.ts

```diff
@@ -396,7 +397,12 @@ export class AuthorizationService {
 					for (const field of requiredPermissions[collection]) {
 						if (field.startsWith('$FOLLOW')) continue;
 						const fieldName = stripFunction(field);
-						if (!allowedFields.includes(fieldName)) {
+
+						if (collection === rootCollection && alias && alias[fieldName]) {
+							if (!allowedFields.includes(alias[fieldName])) {
+								throw new ForbiddenException();
+							}
+						} else if (!allowedFields.includes(fieldName)) {
```

Comment:
Lets clean this up a hair. Couple thoughts:

- Instead of `alias`, lets call it `aliasMap`, so it's a little more obvious what it is
- The `throw` already prevents the next if block from running, so prefer:

```js
if (a === true) {
  throw x;
}

if (b === true) {
  throw y;
}
```

over

```js
if (a === true) {
  throw x;
} else {
  throw y;
}
```

- Lets combine the double nested if

```js
if (a) {
  if (b) {
    // do x
  }
}
```

to 

```js
if (a && b) {
  // do x
}
```

- lets use optional chaining: `alias && alias[fieldName]` -> `alias?.[fieldName]`

🧹

Reply:
> * Instead of `alias`, lets call it `aliasMap`, so it's a little more obvious what it is

Definitely! 👍 

> * The `throw` already prevents the next if block from running

Thanks for the detailed examples of your preferences. It's currently done in this manner as the second block should be skipped if the top block doesn't throw, thus the else block and nested ifs. Your suggested changes should work if the top block continues instead of throwing. 🤔 
I've refactored it to be more similar to eliminate the need for two throw blocks instead.

>  * lets use optional chaining: `alias && alias[fieldName]` -> `alias?.[fieldName]`

Made use of the optional chaining when refactoring. 👍